src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -579,6 +580,21 @@ public bool IsTupleCompatible()
         }
 
         /// <summary>
+        /// Verify if the given type is a tuple of a given cardinality, or can be used to back a tuple type 
+        /// with the given cardinality. 
+        /// </summary>
+        public bool IsTupleOrCompatibleWithTupleOfCardinality(int targetCardinality)
```

Comment:
I see that `IsTupleCompatible` is now for types that could be upgraded to tuples, but are not yet tuples. If so, you could simplify name for `IsTupleOrCompatibleWithTupleOfCardinality` to `IsTupleOrTupleCompatible`.